src/pybind/mgr/mds_partitioner/fs/scheduler.py

```diff
@@ -0,0 +1,166 @@
+from .exception import MDSPartException
+from .analyzer import Analyzer, AnalyzerState
+from .mover import Mover
+from enum import Enum, unique
+import logging
+import threading
+import time
+import json
+
+log = logging.getLogger(__name__)
+
+@unique
+class SchedulerState(Enum):
+    INITIALIZED          = 0
+    INVOKED              = 1
+    RUNNING_ANALYZER     = 2
+    RUNNING_MOVER        = 3
+    WAIT                 = 4
+    STOP                 = 5
+
+class Scheduler:
+    def __init__(self, analyzer, mover, policy_mgr, period=3600):
+        self.analyzer = analyzer
+        self.mover = mover
+        self.policy_mgr = policy_mgr
+        self.scheduler_period = period
+        self.analyzer_period = 30
```

Comment:
So, this is pretty much hard-coded and I should ask if 30s is enough for the analyser for the analyzer to decide on a partition?

Reply:
I think so. The time is fixed too short. Those values can be adjusted by parameter passing or ceph config set.

Reply:
Parameters can be passed.
https://github.com/ceph/ceph/pull/54263/files#diff-5e25a13357d13efe4d4032b130722aaad8f750b216f6689dae5107497e74cd12R198